reVX/pras/rev_reeds_pras.py

```diff
@@ -0,0 +1,690 @@
+
+        logger.info('Saving profiles for {}_{} to file: {}'
+                    .format(self._tech_type, self._res_class, out_fpath))
+
+        profiles = self.tz_convert_profiles(profiles, self._timezone)
+
+        with Outputs(out_fpath, mode='a') as out:
+            out['generators/capacity'][:, self.pras_indices] = \
+                profiles[:, self.sc_build_indices]
+
+        names = np.arange(profiles.shape[1])
+        if self._plant_name_col is not None:
+            names = self.get_unique_plant_names(meta, self._plant_name_col,
+                                                self._tech_tag)
+
+        df_plx = pd.DataFrame(profiles, columns=names,
+                              index=time_index.tz_convert(None))
+        df_plx.index.name = 'DATETIME'
+        csv_fp = out_fpath.replace('.h5', '_{}_{}.csv'.format(self._tech_type,
+                                                              self._res_class))
+        df_plx.to_csv(csv_fp)
+
+        logger.info('Wrote pras formatted profiles to: {}'.format(csv_fp))
+
+    @classmethod
+    def write_load_data(cls, load_path, out_fpath):
+        """Export load data to pras file
+
+        Parameters
+        ----------
+        load_path : str
+            Path to load data to use for overwriting 'regions/load' dataset
+            in out_fpath.
+        out_fpath : str
+            Path to .h5 file into which plant buildout should be saved. A
+            pras-formatted csv will also be written in the same directory.
+        """
+        if not out_fpath.endswith('.h5'):
+            out_fpath = out_fpath + '.h5'
+
+        msg = ("out_fpath={} does not exist. This file must come from a "
+               "previous ReEDS2PRAS run").format(out_fpath)
+        assert os.path.exists(out_fpath), msg
+
+        new_out_file = out_fpath.replace('.h5', '_updated.h5')
+        if not os.path.exists(new_out_file):
+            shutil.copy(out_fpath, new_out_file)
+        out_fpath = new_out_file
+
+        logger.info('Saving load from {} to {}'.format(load_path, out_fpath))
+
+        with Outputs(out_fpath, mode='a') as out:
+            with Resource(load_path) as res:
+                out['regions/load'] = res['load']
+
+    def _make_node_map(self):
+        """Map built rev SC points to pras nodes.
+
+        Returns
+        -------
+        pras_node_index : np.ndarray
+            (n, 1) array of pras node indices mapped to the SC builds where n
+            is the number of SC points built.  Each value in this array gives
+            the pras node index that the sc point is mapped to. So
+            self.node_map[10] yields the pras node index for
+            self.sc_build[10].
+        """
+        if 'region' in self.sc_build:
+            pras_node_index = self.sc_build['region'].values
+            pras_node_index = [self.region_node_map[k] for k
+                               in pras_node_index]
+            pras_node_index = np.array(pras_node_index)
```

Comment:
So yeah, as discussed we're actually using the buildout file for the "plexos nodes" input right? This was just surprising, I expected the reeds zonal shape file to be required, and we should update the "plexos_nodes" docstrings or remove entirely if its not necessary.